Add user notification

WNjihia opened the pull request:
Closes issue #5 
When a data request is added, organization members are notified via email.

DavidLemayian commented on ckanext/datarequests/controllers/ui_controller.py:
```diff
@@ -174,7 +176,28 @@ def _process_post(self, action, context):
+                    }
+                    subject = base.render_jinja2('emails/notify_user_subject.txt',
+                                                 extra_vars)
+                    subject = subject.split('\n')[0]
```
Why are we splitting here?

WNjihia replied:
Apologies. I have just realized that I don't need to split. I will make the necessary changes

WNjihia replied:
Done

DavidLemayian commented on ckanext/datarequests/controllers/ui_controller.py:
```diff
@@ -174,7 +176,28 @@ def _process_post(self, action, context):
+
+                    for user in users:
+                        user_data = model.User.get(user['id'])
+                        extra_vars['user_name'] = user_data.fullname
```
This should probably be called `user_fullname`

WNjihia replied:
It's done. I have changed it to user_fullname

DavidLemayian commented:
Thanks @WNjihia. Documentation needs to now be updated on `README.md`.

DavidLemayian commented:
Let's also add any relevant doc strings.

WNjihia commented:
@DavidLemayian Since the user will automatically be notified when a new data request is added, I don't think it will be necessary to add that information on the `README.md`

DavidLemayian commented:
@WNjihia The adding of documentation isn't for users but for devs who re-use this plugin or are doing development on it. We'd like to show on the README.md the features that are available and how we've gone about implementation. e.g should they ensure SMTP configs are set in CKAN installation?

Having had to write that out has actually made me ask a few more questions:
1. Have we added test for this?
2. Is this the best place for this code to exist or should be referenced elsewhere instead? This is in fact `ui_controller` and perhaps this functionality should live in `actions.py` instead? Why the choice?

cc @shemogumbe

WNjihia commented:
@DavidLemayian I have updated the README.md. I have also moved the function to `actions.py` and called it in the `ui_controller.py`.
Regarding the tests, could it be done as its own separate task?

WNjihia commented:
@DavidLemayian I think this feature should be added in ckanext-notify. This is because, if merged to the ckanext-datarequests extension, every organization will be required to use email notifications whereas in ckanext-notify, they will be given an option of which notification channel to use

DavidLemayian commented:
@WNjihia completely agree on all the points. This is awesome work. I'll close this and wait on your PR on ckanext-notify instead. :)
